### PR TITLE
Refactor examples/pytorch/NanoGPT/

### PR DESCRIPTION
**Pull request by Filtee**
### What changes were proposed in this pull request?

Refactored example/pytorch/NanoGPT.

### Why are the changes needed?

The original code for NanoGPT example is a bit messy, and it has a hidden little “bug”: it doesn't support gradient accumulation strategy, despite dedicating a considerable amount of space to attempting it.

### Does this PR introduce any user-facing change?

Nope.

### How was this patch tested?

It works fine…

### CODE REVIEW

**Comment by workingloong** [comment]
You should run the pre-commit to check code style with the [tutorial](https://github.com/intelligent-machine-learning/dlrover/blob/master/docs/developer_guide.md#submit-a-pr)

**Comment by Filtee** [comment]
> You should run the pre-commit to check code style with the [tutorial](https://github.com/intelligent-machine-learning/dlrover/blob/master/docs/developer_guide.md#submit-a-pr)

Sorry, I forgot to push the changes after running pre-commit.

**Review by workingloong — APPROVED** [review]
LGTM
